feat: Add Alpha Vantage and Tencent Cloud Search engines

markqiu opened the pull request:
## What does this PR do?

This PR adds two new search engines to SearXNG:

1. **Alpha Vantage Engine** - Financial data and stock symbol search
   - Handles Alpha Vantage's special field name format ("1. symbol", "2. name", etc.)
   - Supports stock symbol and company name search
   - Returns symbol, type, region, currency, and match score

2. **Tencent Cloud Search Engine** - High-quality Chinese web search
   - Implements TC3-HMAC-SHA256 signature authentication
   - Supports multiple search modes (natural, multimodal VR, mixed)
   - Site-specific and industry filtering capabilities
   - Optimized for Chinese language content

## Why is this change important?

1. Expands SearXNG's financial search capabilities with stock/company lookup
2. Significantly improves Chinese language search quality
3. Both engines use official APIs with proper authentication
4. No new dependencies required
5. Both engines are production-ready with comprehensive testing

## How to test this PR locally?

```bash
# Run unit tests (no API keys required)
./manage test.unit

# Specific engine tests:
./manage pyenv.cmd python -m nose2 tests.unit.engines.test_alphavantage
./manage pyenv.cmd python -m nose2 tests.unit.engines.test_tencent

# Configure API keys in settings.yml (see utils/tencent_settings_example.yml)
# Then test with real APIs:
./manage test.custom_engines http://localhost:8888

# Or manually:
curl "http://localhost:8888/search?q=AAPL&format=json&engines=alphavantage"
curl "http://localhost:8888/search?q=北京天气&format=json&engines=tencent"
```

## Author's checklist

- [x] Unit tests: 14 tests, all passing (6 for Alpha Vantage, 8 for Tencent)
- [x] Code quality: Pylint 10.00/10 and 9.99/10
- [x] Code formatting: Black formatted
- [x] Documentation: Complete RST docs with Sphinx integration
- [x] SPDX license headers added
- [x] Both engines disabled by default
- [x] No breaking changes
- [x] No new dependencies

## Related issues

N/A - New feature addition

markqiu committed:
feat: Add Alpha Vantage and Tencent Cloud Search engines

- Add Alpha Vantage engine for stock/financial data search
  * Handles special field names ("1. symbol", "2. name", etc.)
  * Supports stock symbol and company name search
  * Returns symbol, type, region, currency, and match score

- Add Tencent Cloud Search engine for Chinese web search
  * Implements TC3-HMAC-SHA256 signature authentication
  * Supports multiple search modes (natural, multimodal VR, mixed)
  * Supports site-specific and industry filtering
  * Optimized for Chinese language content

- Add comprehensive unit tests
  * 6 tests for Alpha Vantage engine
  * 8 tests for Tencent engine
  * All tests use Mock objects (no API keys required)
  * Follows SearxTestCase pattern
  * All 14 tests passing

- Add RST documentation integrated with Sphinx
  * Complete configuration guides
  * Usage examples and troubleshooting
  * API documentation links

- Add test.custom_engines function to test framework
  * Integrated into utils/lib_sxng_test.sh
  * Accessible via ./manage test.custom_engines

- Add Tencent API direct test tool
  * utils/test_tencent_direct.py for credential verification

- Update settings.yml with engine configurations
  * Both engines disabled by default
  * Detailed configuration comments

Code quality:
- Pylint: 10.00/10 and 9.99/10
- Black formatted
- SPDX license headers
- Complete docstrings
- Following SearXNG naming conventions (alphavantage.py, tencent.py)

return42 commented:
> Expands SearXNG's financial search capabilities with stock/company lookup

Thanks a lot :+1: 

> Add Alpha Vantage and Tencent Cloud Search engines

Can we create one PR for each engine?  / Lets start with one PR and when this PR is merged, we can review the other PR.. 

BTW project language is English, please translate docs and comments to English.

markqiu committed:
refactor: Update comments and documentation in Alpha Vantage and Tencent engines

- Translated comments from Chinese to English for better clarity and accessibility.
- Improved documentation strings to enhance understanding of functions and parameters.
- Ensured consistency in comment style across both engines for better readability.
